fraud/bad_encoding.go

```diff
@@ -71,6 +71,12 @@ func (p *BadEncodingProof) MarshalBinary() ([]byte, error) {
 	return badEncodingFraudProof.Marshal()
 }
 
+// UnmarshalBEFP converts given data to BadEncodingProof
+func UnmarshalBEFP(data []byte) (Proof, error) {
+	befp := &BadEncodingProof{}
+	return befp, befp.UnmarshalBinary(data)
```

Comment:
Yes, but does not it modify in-place befp every time we call `befp.UnmarshalBinary(data)`?

Reply:
It does modify it.

Reply:
You can use append like this in some cases, if you know the append rules.

Reply:
It does modify it in place, it's just really not-obvious by looking at it. Knowing that is extra cognitive overhead that's not needed because you can just split it up into another line.

Reply:
> Note that we've used append incorrectly in our own code in the past. That alone should disqualify return befp, befp.UnmarshalBinary(data).

I disagree here. The fact that we used _tabs_ incorrectly does not disqualify the usage of _spaces_.  Though I am not feeling strong about an additional line or not, I just think append is a different case, though it has common _subjective_ cognitive overhead property.

When I see something that requires me to put in additional cognitive efforts, my response to this is usually: "wow, that's cool or smart". This is why I meant subjective.

Reply:
> When I see something that requires me to put in additional cognitive efforts, my response to this is usually: "wow, that's cool or smart". This is why I meant subjective.

Without judging the above sample at hand: in general this might be cool or smart from an intellectual point of view, but we should aim for the most readable code without cognitive overhead (even if that is often boring).